Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -389,66 +389,111 @@ ErrorOr<NonnullRefPtr<Gfx::Bitmap>> Bitmap::scaled(float sx, float sy) const
+
+                auto a = get_pixel(i, j);
+                auto b = get_pixel(i + 1, j);
+                auto c = get_pixel(i, j + 1);
+                auto d = get_pixel(i + 1, j + 1);
+
+                auto e = a.interpolate(b, u);
+                auto f = c.interpolate(d, u);
+                auto color = e.interpolate(f, v);
+                new_bitmap->set_pixel(x, y, color);
+            }
+        }
+
+        // Bottom strip (excluding last pixel)
+        auto old_bottom_y = old_height - 1;
+        auto new_bottom_y = new_height - 1;
         for (int x = 0; x < new_width - 1; x++) {
             auto p = static_cast<float>(x) * static_cast<float>(old_width - 1) / static_cast<float>(new_width - 1);
-            auto q = static_cast<float>(y) * static_cast<float>(old_height - 1) / static_cast<float>(new_height - 1);
 
             int i = floorf(p);
-            int j = floorf(q);
             float u = p - static_cast<float>(i);
+
+            auto a = get_pixel(i, old_bottom_y);
+            auto b = get_pixel(i + 1, old_bottom_y);
+            auto color = a.interpolate(b, u);
+            new_bitmap->set_pixel(x, new_bottom_y, color);
+        }
+
+        // Right strip (excluding last pixel)
+        auto old_right_x = old_width - 1;
+        auto new_right_x = new_width - 1;
+        for (int y = 0; y < new_height - 1; y++) {
+            auto q = static_cast<float>(y) * static_cast<float>(old_height - 1) / static_cast<float>(new_height - 1);
+
+            int j = floorf(q);
             float v = q - static_cast<float>(j);
 
-            auto a = get_pixel(i, j);
-            auto b = get_pixel(i + 1, j);
-            auto c = get_pixel(i, j + 1);
-            auto d = get_pixel(i + 1, j + 1);
+            auto c = get_pixel(old_right_x, j);
+            auto d = get_pixel(old_right_x, j + 1);
 
-            auto e = a.interpolate(b, u);
-            auto f = c.interpolate(d, u);
-            auto color = e.interpolate(f, v);
-            new_bitmap->set_pixel(x, y, color);
+            auto color = c.interpolate(d, v);
+            new_bitmap->set_pixel(new_right_x, y, color);
         }
-    }
 
-    // Bottom strip (excluding last pixel)
-    auto old_bottom_y = old_height - 1;
-    auto new_bottom_y = new_height - 1;
-    for (int x = 0; x < new_width - 1; x++) {
-        auto p = static_cast<float>(x) * static_cast<float>(old_width - 1) / static_cast<float>(new_width - 1);
+        // Bottom-right pixel
+        new_bitmap->set_pixel(new_width - 1, new_height - 1, get_pixel(physical_width() - 1, physical_height() - 1));
 
-        int i = floorf(p);
-        float u = p - static_cast<float>(i);
+    } else if (old_height == 1 && old_width >= 1) {
+        // Copy horizontal strip multiple times (excluding last pixel)
+        auto old_bottom_y = old_height - 1;
+        for (int new_bottom_y = 0; new_bottom_y < new_height; new_bottom_y++) {
+            for (int x = 0; x < new_width - 1; x++) {
```

Comment:
You should be able to swap these loops to interpolate the pixels only once per row/column, and set the final row/column to the last pixel in another loop.

Reply:
I couldn't figure out how swapping the loops alone would help, but since interpolation is only required to be done once for a row I now copy the interpolated value into all columns at once. I think that was what you meant with your comment.
```
        for (int x = 0; x < new_width - 1; x++) {
            auto p = static_cast<float>(x) * static_cast<float>(old_width - 1) / static_cast<float>(new_width - 1);
            int i = floorf(p);
            float u = p - static_cast<float>(i);

            auto a = get_pixel(i, old_bottom_y);
            auto b = get_pixel(i + 1, old_bottom_y);
            auto color = a.interpolate(b, u);
            for (int new_bottom_y = 0; new_bottom_y < new_height; new_bottom_y++) {
                // Interpolate color only once and then copy into all columns.
                new_bitmap->set_pixel(x, new_bottom_y, color);
            }
        }
```

Reply:
That's exactly what I meant, thank you! sorry if my explanation was considering.